# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/Document.cpp
# Commit: 8705284b50d4
# Full Hash: 8705284b50d4905110df9b1fab7f00d9d77d06e6
# Author: Christoph Kerschbaumer <ckerschb@christophkerschbaumer.com>
# Date: 2019-10-22 15:53:00
# Regressor Bug: 1584993
# File Overlap Count: 1
# Description:
#   Bug 1584993: Make CSP frame-ancestors work with fission enabled. r=jkt,farre,valentin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D49147
# ==============================================================================

diff -r 141755ff04e1 -r 8705284b50d4 dom/base/Document.cpp
--- a/dom/base/Document.cpp	Tue Oct 22 08:56:13 2019 +0000
+++ b/dom/base/Document.cpp	Tue Oct 22 08:53:47 2019 +0000
@@ -3286,21 +3286,6 @@
     SetPrincipals(principal, principal);
   }
 
-  // ----- Enforce frame-ancestor policy on any applied policies
-  nsCOMPtr<nsIDocShell> docShell(mDocumentContainer);
-  if (docShell) {
-    bool safeAncestry = false;
-
-    // PermitsAncestry sends violation reports when necessary
-    rv = mCSP->PermitsAncestry(docShell, &safeAncestry);
-
-    if (NS_FAILED(rv) || !safeAncestry) {
-      MOZ_LOG(gCspPRLog, LogLevel::Debug,
-              ("CSP doesn't like frame's ancestry, not loading."));
-      // stop!  ERROR page!
-      aChannel->Cancel(NS_ERROR_CSP_FRAME_ANCESTOR_VIOLATION);
-    }
-  }
   ApplySettingsFromCSP(false);
   return NS_OK;
 }